lib/ansible/modules/system/group.py

```diff
@@ -143,11 +145,17 @@ def group_mod(self, **kwargs):
         return self.execute_command(cmd)
 
     def group_exists(self):
-        try:
-            if grp.getgrnam(self.name):
-                return True
-        except KeyError:
+        if self.local:
+            for line in get_file_content(self.GROUPFILE, '').splitlines():
```

Comment:
This is making a bad assumption, while most of the time the groups file is the way people configure local users, it is the default, NOT the only way. 

You can setup a local db or other source of users/groups on the system, it is not just the group file + ldap/nis/other network service.

Reply:
Agree there are other setups (but group and passwd files is the standard on most Distros); but my solution is a improvement on the current code. And as i do not have access to others setups, this is the best i can do. 

While it won't fix all possible setups in the world it will fix most (i hope) and so i hope it gets accepted as is.

Reply:
@bcoca ship it as is ?

Reply:
no, the other issue is that gshadow and other 'sources' (nss_db?) are still left with possible dangling data. I've been trying to find a good solution that handles all the 'niceties' of nsswitch